components/places/sql/create_sync_triggers.sql

```diff
@@ -0,0 +1,171 @@
+-- This Source Code Form is subject to the terms of the Mozilla Public
+-- License, v. 2.0. If a copy of the MPL was not distributed with this
+-- file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+-- This file defines triggers for the Sync connection.
+
+--- Pushes uploaded changes back to the local and remote trees. This is more
+--- or less equivalent to Desktop's `PlacesSyncUtils.bookmarks.pushChanges`.
+CREATE TEMP TRIGGER pushUploadedChanges
+AFTER UPDATE OF uploadedAt ON itemsToUpload WHEN NEW.uploadedAt > -1
+BEGIN
+  -- Reduce the change counter and update the sync status for uploaded items.
+  -- If the item was uploaded during the sync, its change counter will still
+  -- be > 0 for the next sync.
+  UPDATE moz_bookmarks SET
```

Comment:
Nit: We use 4 spaces to indent trigger bodies elsewhere so should here too IMO.

Also I don't seem to be able to see all my comments so sorry if I say this multiple times or whatever